share/dbtypes_gen.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+import os
+import sys
+import json
+
+with open(os.path.join(os.path.dirname(__file__), '..', 'src', 'query_backend.h')) as f:
+    lines = f.readlines()
+
+VERSION = "v"+sys.argv[1]
+
+ALL_BACK = ["SQLite", "HDF5"]
+
+ENUM_START = 0
+
+for i in range(0,len(lines)):
+    if "enum DbTypes {" in lines[i]:
+        ENUM_START = i
+
+ID_INDEX = 0
+DB_INDEX = 1
+CPP_INDEX = 2
+SHAPE_INDEX = 3
+BACK_INDEX = 4
+VERSION_INDEX = 5
+SUPPORTED_INDEX = 6
+CANON_INDEX = 7
+VL_INDEX = 8
+
+def main():
+    outer = []
+    for back in ALL_BACK:
+        ID = 0
+        inner = []
+        for i in range(ENUM_START+1,len(lines)):
+            s = lines[i].split("//")[-1].strip()
+            #The extra strip statements keep the " = 0" out of the BOOL dbtype name.
+            DB = lines[i].split("//")[0].strip().strip(",").strip("0").strip(" =")
+            if "};" in lines[i]:
+                break
+            try:
+                x = json.loads(s)
+                x.insert(ID_INDEX, ID)
+                x.insert(DB_INDEX, DB)
+                supported_backends = x[BACK_INDEX]
+                x[BACK_INDEX] = back
+                x.insert(VERSION_INDEX,VERSION)
+                x.insert(SUPPORTED_INDEX, 1 if back in supported_backends else 0)
+                inner.append(x)
+                ID += 1
+            except ValueError:
```

Comment:
Only the line that actually causes the value error should be in the try block.  If DB is empty, then you should skip this block.